logstash-core/lib/logstash/plugin_metadata.rb

```diff
@@ -29,9 +29,7 @@ module LogStash
   class PluginMetadata
     include LogStash::Util::Loggable
 
-    Thread.exclusive do
```

Comment:
cool, had done this locally introducing a mutex (explicitly) to sync against.
was really confused why the `exclusive` was necessary in the first place : 
to avoid potential double initialization of the var as 2 threads are loading the this.rb?
not sure that is possible but the code indicates it might have been the case ...

Reply:
right I don't see how a race condition can happen here, maybe @yaauie can chime in there.

This is also why I switched to a CONSTANT, so we'll see a warning if this really happens.

Reply:
IIRC, it was protection against double initialisation, but unnecessary on the versions of jruby we use.

👍 for the solution.